implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpHttpMetricsSender.java

```diff
@@ -49,17 +49,16 @@ public void send(Request request) throws Exception {
             .withHeader("User-Agent", userAgentHeader)
             .withContent("application/x-protobuf", request.getMetricsData());
         request.getHeaders().forEach(httpRequest::withHeader);
-        HttpSender.Response response;
         try {
-            response = httpRequest.send();
+            HttpSender.Response response = httpRequest.send();
+            if (!response.isSuccessful()) {
+                throw new OtlpHttpMetricsSendUnsuccessfulException(String
```

Comment:
I think this will be caught by the `catch` block below and thrown forward wrapped in an `Exception` while prior this change, an `OtlpHttpMetricsSendUnsuccessfulException` was thrown.

Reply:
@jonatan-ivanov Oops, sorry. I reverted it.